File: GuessNumber8/app/src/main/java/com/android/guessnumber8/MainActivity.java
package com.android.guessnumber8;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GestureDetectorCompat;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.gesture.Gesture;
import android.gesture.GestureLibraries;
import android.gesture.GestureLibrary;
import android.gesture.GestureOverlayView;
import android.gesture.Prediction;
import android.os.Bundle;
import android.util.Log;
import android.view.GestureDetector;
import android.view.MotionEvent;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Random;

public class MainActivity extends AppCompatActivity implements GestureOverlayView.OnGesturePerformedListener {

    private static final String LOG_TAG =
            MainActivity.class.getSimpleName();

    private GestureLibrary gLibrary;
    private GestureDetectorCompat mDetector;

    private TextView tvInfo;
    private EditText etInput;
    private Button bControl;
    private Button newGameButton;
    private int number;
    private ProgressBar progressBar;
    private int progressStatus = 0;
    private boolean isOver;
    AlertDialog.Builder builder;
    StringBuffer input;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mDetector = new GestureDetectorCompat(this, new MyGestListener());

        gLibrary =
                GestureLibraries.fromRawResource(this, R.raw.gestures);
        if (!gLibrary.load()) {
            finish();
        }

        GestureOverlayView gOverlay =
                (GestureOverlayView) findViewById(R.id.gOverlay);
        gOverlay.addOnGesturePerformedListener(this);


        tvInfo = findViewById(R.id.textView);
        etInput = findViewById(R.id.editText);
        bControl = findViewById(R.id.button);
        newGameButton = findViewById(R.id.button2);
        progressBar = findViewById(R.id.progressBar);
        builder = new AlertDialog.Builder(MainActivity.this);

        newGame();
    }

    @Override
    public boolean onTouchEvent(MotionEvent event){
        this.mDetector.onTouchEvent(event);
        return super.onTouchEvent(event);
    }
    class MyGestListener extends GestureDetector.SimpleOnGestureListener {
        @Override
        public
        boolean onFling(MotionEvent event1, MotionEvent event2,
                        float velocityX, float velocityY)
        //void onLongPress(MotionEvent event)
        {
            //tvOutput.setText("onLongPress: " + event.toString());
            Log.e(LOG_TAG, String.format("stop app"));
            finish();
            System.exit(0);
            return true;
        }
    }



    public void onGesturePerformed(GestureOverlayView overlay, Gesture
            gesture) {
        ArrayList<Prediction> predictions =
                gLibrary.recognize(gesture);

        if (predictions.size() > 0 && predictions.get(0).score > 1.0) {

            String action = predictions.get(0).name;
            Toast.makeText(this, action, Toast.LENGTH_SHORT).show();


            if (predictions.get(0).name.equals("one"))
                input.append("1");
             else if (predictions.get(0).name.equals("two")) {
                input.append("2");
            } else if (predictions.get(0).name.equals("three")) {
                input.append("3");
            } else if (predictions.get(0).name.equals("four")) {
                input.append("4");
            } else if (predictions.get(0).name.equals("five")) {
                input.append("5");
            } else if (predictions.get(0).name.equals("six")) {
                input.append("6");
            } else if (predictions.get(0).name.equals("seven")) {
                input.append("7");
            } else if (predictions.get(0).name.equals("eight")) {
                input.append("8");
            } else if (predictions.get(0).name.equals("nine")) {
                input.append("9");
            } else if (predictions.get(0).name.equals("zero")) {
                input.append("0");
            } else if (predictions.get(0).name.equals("stop"))
            {
                etInput.setText(input.toString());
                Log.e(LOG_TAG, String.format("stop gesture %s", input));
            }
        }

    }


    private void newGame()
    {
        Log.e(LOG_TAG, String.format("new game"));
        progressStatus = 0;
        progressBar.setProgress(progressStatus);
        Random r = new Random();
        int low = 1;
        int high = 10; // 200, 10 to test
        number = r.nextInt(high - low) + low;
        newGameButton.setEnabled(false);
        bControl.setEnabled(true);
        tvInfo.setText(R.string.try_to_guess);
        isOver = false;
        input = new StringBuffer();

    }

    public void enterButton(View view) {
        Log.e(LOG_TAG, String.format("Enter button"));

        input = new StringBuffer();

        progressStatus++;
        if (progressStatus > 10)
        {
            builder.setTitle(getResources().getString(R.string.oops))
                    .setMessage(getResources().getString(R.string.tries_over))
                    .setCancelable(false)
                    .setNegativeButton("OK",
                            new DialogInterface.OnClickListener() {
                                public void onClick(DialogInterface dialog, int id) {
                                    isOver = true;
                                    newGameButton.setEnabled(true);
                                    dialog.cancel();
                                }
                            });
            AlertDialog alert = builder.create();
            alert.show();
        }
        else {

            progressBar.setProgress(progressStatus);

            if (etInput.getText().toString().equals("")) {
                Log.e(LOG_TAG, String.format("Empty input"));

                builder.setTitle(R.string.error)
                        .setMessage(R.string.empty)
                        .setCancelable(false)
                        .setNegativeButton("OK",
                                new DialogInterface.OnClickListener() {
                                    public void onClick(DialogInterface dialog, int id) {
                                        dialog.cancel();
                                    }
                                });
                AlertDialog alert = builder.create();
                alert.show();
            } else {

                try {
                    int temp = Integer.parseInt(etInput.getText().toString());

                    if (temp < 1 || temp > 10) {
                        builder.setTitle(R.string.error)
                                .setMessage(R.string.value_borders)
                                .setCancelable(false)
                                .setNegativeButton("OK",
                                        new DialogInterface.OnClickListener() {
                                            public void onClick(DialogInterface dialog, int id) {
                                                dialog.cancel();
                                            }
                                        });
                        AlertDialog alert = builder.create();
                        alert.show();

                        etInput.setText("");
                    } else {
                        if (temp > number) {
                            tvInfo.setText(getResources().getString(R.string.ahead));
                        } else if (temp < number) {
                            tvInfo.setText(getResources().getString(R.string.behind));
                        } else if (temp == number) {
                            tvInfo.setText(getResources().getString(R.string.hit));

                            builder.setTitle(getResources().getString(R.string.hit))
                                    .setMessage(getResources().getString(R.string.hit) + " in " + progressStatus + " tries")
                                    .setCancelable(false)
                                    .setNegativeButton("OK",
                                            new DialogInterface.OnClickListener() {
                                                public void onClick(DialogInterface dialog, int id) {
                                                    isOver = true;
                                                    newGameButton.setEnabled(true);
                                                    bControl.setEnabled(false);
                                                    //newGame();
                                                    //etInput.setText("");

                                                    dialog.cancel();
                                                }
                                            });
                            AlertDialog alert = builder.create();
                            alert.show();


                        }
                    }
                } catch (NumberFormatException e) {
                    builder.setTitle(R.string.error)
                            .setMessage(R.string.value_not_int)
                            .setCancelable(false)
                            .setNegativeButton("OK",
                                    new DialogInterface.OnClickListener() {
                                        public void onClick(DialogInterface dialog, int id) {
                                            dialog.cancel();
                                        }
                                    });
                    AlertDialog alert = builder.create();
                    alert.show();

                    etInput.setText("");
                }
            }
        }
    }

    public void newGame(View view) {
        newGame();
        etInput.setText("");

        //image animation
    }
}

File: Calculator8/app/src/main/java/com/android/calculator8/MainActivity.kt
package com.android.calculator8

import android.content.DialogInterface
import android.gesture.Gesture
import android.gesture.GestureLibraries
import android.gesture.GestureLibrary
import android.gesture.GestureOverlayView
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.activity_main.*
import java.util.ArrayList

class MainActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener,  GestureOverlayView.OnGesturePerformedListener{

    var listOfItemsFrom = arrayListOf("", "")
    var listOfItemsTo = arrayListOf("", "")

    private var gLibrary: GestureLibrary? = null
    internal var input = StringBuffer()
    private var etInput: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        gLibrary = GestureLibraries.fromRawResource(this, R.raw.gestures)
        if (!gLibrary!!.load()) {

            finish()
        }

        val gOverlay = findViewById<GestureOverlayView>(R.id.gOverlay)
        gOverlay.addOnGesturePerformedListener(this)
        etInput = findViewById(R.id.editText_from)


        listOfItemsFrom = resources.getStringArray(R.array.listOfItemsFrom).toCollection(ArrayList<String>())
        listOfItemsTo = resources.getStringArray(R.array.listOfItemsTo).toCollection(ArrayList<String>())



        // Create an ArrayAdapter using a simple spinner layout and languages array
        val aa1 = ArrayAdapter(this, android.R.layout.simple_spinner_item, listOfItemsFrom)
        // Set layout to use when the list of choices appear
        aa1.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        // Set Adapter to Spinner
        spinner_from!!.adapter = aa1

        val aa2 = ArrayAdapter(this, android.R.layout.simple_spinner_item, listOfItemsTo)
        aa2.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner_to!!.adapter = aa2
    }

    override fun onGesturePerformed(overlay: GestureOverlayView, gesture: Gesture) {
        etInput?.setText("")

        val predictions = gLibrary!!.recognize(gesture)

        if (predictions.size > 0 && predictions[0].score > 1.0) {

            val action = predictions[0].name
            Toast.makeText(this, action, Toast.LENGTH_SHORT).show()


            if (predictions[0].name == "one")
                input.append("1")
            else if (predictions[0].name == "two") {
                input.append("2")
            } else if (predictions[0].name == "three") {
                input.append("3")
            } else if (predictions[0].name == "four") {
                input.append("4")
            } else if (predictions[0].name == "five") {
                input.append("5")
            } else if (predictions[0].name == "six") {
                input.append("6")
            } else if (predictions[0].name == "seven") {
                input.append("7")
            } else if (predictions[0].name == "eight") {
                input.append("8")
            } else if (predictions[0].name == "nine") {
                input.append("9")
            } else if (predictions[0].name == "zero") {
                input.append("0")
            } else if (predictions[0].name == "stop") {
                etInput?.setText(input.toString())
                //Log.e(LOG_TAG, String.format("stop gesture %s", input))
            }
        }

    }


    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        if(parent.id == R.id.spinner_from)
        {
            from.text = listOfItemsFrom[position]
            editText_from.isEnabled = true
        }
        else if(parent.id == R.id.spinner_to)
        {
            to.text = listOfItemsTo[position]
        }
    }

    override fun onNothingSelected(parent: AdapterView<*>) {
        if(parent.id == R.id.spinner_from)
        {
            from.setText(R.string.from)
            editText_from.isEnabled = false
        }
        else if(parent.id == R.id.spinner_to)
        {
            to.setText(R.string.to)
        }
    }


    private val positiveButtonClick = { dialog: DialogInterface, which: Int ->
    }

    private fun basicAlert(view: View){

        val builder = AlertDialog.Builder(this)

        with(builder)
        {
            setTitle(R.string.error)
            setMessage(R.string.wrong_input)
            setPositiveButton("OK", DialogInterface.OnClickListener(function = positiveButtonClick))
            show()
        }


    }

    fun calcButton(view: View) {
        try {
            var fromVal:Double = editText_from.text.toString().toDouble()
            if (fromVal >= 0) {
                if (spinner_from.selectedItem.equals(listOfItemsFrom[1]))
                    fromVal /= 1000000.0

                if (spinner_to.selectedItem.equals(listOfItemsTo[0])) {
                    editText_to.text = (fromVal * 10763910.0).toString()
                } else {
                    editText_to.text = (fromVal * 247.1).toString()
                }
            }
            else
            {
                basicAlert(view)
            }
        } catch (e: NumberFormatException) {
            basicAlert(view)
        }
        input = StringBuffer()
    }


}
